src/utilities/include/vec_dh.h

```diff
@@ -17,6 +17,13 @@
 #include <cuda.h>
 #endif
 
+#if TRACY_ENABLE
+#include <tracy/Tracy.hpp>
+#else
+#define TracyAllocS(ptr, size, n) (void)0
+#define TracyFreeS(ptr, n) (void)0
+#endif
```

Comment:
The whole point of VecDH (at least originally) was to handle the fact that CUDA needs separate host and device vectors that need to be synced. With CUDA now removed, is there a chance we can remove VecDH entirely? Doesn't have to be in this PR, but it would be great if we can simplify here.

Reply:
Yes this is what I am going to do, but I plan to do this in another PR as that will be a lot of changes.

Reply:
Thinking about it, I'm thinking if we may want to have vulkan acceleration later. Vulkan would require something similar or perhaps a bit more complicated (as we may not have unified memory, depending on the target hardware).

Reply:
also, as our `VecDH` class has specialized parallel fill and uninitialized memory routines, changing to `std::vector<T>` can result in more than 20% performance regression for some test cases.

Reply:
I have no problem with leaving the class if it's actually doing something useful, but if it can be simplified to one internal storage instead of two, that would be great. We can always redesign it later if needed for Vulkan, but I don't want to plan early for that, especially considering that TBB ended up being just as fast as CUDA. That makes me think Vulkan may be more trouble than it's worth.

Reply:
Yeah I can simplify that in another PR. It was really surprising when I see that switching to std::vector causes a large slowdown.

Reply:
Regarding Vulkan, I think the only case where GPU can be faster is SDF.

Reply:
Good to know. Maybe someday SDF will be used enough that that'll be worth looking into, but for now I'm thinking TBB is just fine.